Build fails occasionally at test phase due to surefire issue

Throwing below error,
Waiting for Jenkins to finish collecting data[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project osgi-tests: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test failed: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /build/jenkins-home/workspace/platform-builds/carbon-deployment/tests/osgi-tests && /build/jenkins-home/software/java/jdk1.8.0_45/jre/bin/java '-javaagent:/build/jenkins-home/workspace/platform-builds/carbon-deployment/.repository/org/jacoco/org.jacoco.agent/0.7.5.201505241946/org.jacoco.agent-0.7.5.201505241946-runtime.jar=destfile=/build/jenkins-home/workspace/platform-builds/carbon-deployment/tests/osgi-tests/target/jacoco.exec,includes=org.wso2.carbon*' -jar /build/jenkins-home/workspace/platform-builds/carbon-deployment/tests/osgi-tests/target/surefire/surefirebooter5152829544684532020.jar /build/jenkins-home/workspace/platform-builds/carbon-deployment/tests/osgi-tests/target/surefire/surefire8592043572268409080tmp /build/jenkins-home/workspace/platform-builds/carbon-deployment/tests/osgi-tests/target/surefire/surefire_24347295732707164145tmp

--- a/pom.xml
+++ b/pom.xml
@@ -59,6 +59,16 @@
                     <artifactId>carbon-feature-plugin</artifactId>
                     <version>${carbon.feature.plugin.version}</version>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <version>${maven.surefire.plugin.version}</version>
+                    <configuration>
+                        <systemPropertyVariables>
+                            <jacoco-agent.destfile>target/jacoco.exec</jacoco-agent.destfile>
+                        </systemPropertyVariables>
+                    </configuration>
+                </plugin>
             </plugins>
         </pluginManagement>
 
@@ -408,7 +418,7 @@
         <javax.inject.version>1</javax.inject.version>
         <jacoco.version>0.7.5.201505241946</jacoco.version>
         <org.eclipse.equinox.executable.version>3.5.0.v20110530-7P7NFUFFLWUl76mart</org.eclipse.equinox.executable.version>
-
+        <maven.surefire.plugin.version>2.18.1</maven.surefire.plugin.version>
     </properties>
 
 </project>

